File: rb/dice.rb
class Dice
  def initialize(number_of_faces)
    unless number_of_faces.is_a?(Integer) && number_of_faces > 0
      raise StandardError.new "Error: Please, instantiate the Dice class with a number of Dice faces greater than zero"
    end

    @faces = []
    @last_drawn_face = nil
    number_of_faces.times {
      |i| @faces[i] = i + 1
    }
  end

  def shuffle()
    unless @faces.length > 0
      raise StandardError.new "Error: Cannot shuffle a Dice without faces"
    end
    shuffled_faces = @faces.shuffle()
    return @last_drawn_face = shuffled_faces[0]
  end

  def get_last_draw_face()
    unless @last_drawn_face
      raise StandardError.new "Error: No lastDrawnFace attribute of Dice"
    end
    return @last_drawn_face
  end
end 

File: dice.js
const assert = require('node:assert');
const { getRandomInt } = require('./utils');

class Dice {
    #faces = [];
    #lastDrawnFace = null;

    constructor (numberOfFaces){
        assert(numberOfFaces > 0, 'Error: Please, instantiate the Dice class with a number of Dice faces greater than zero');
        for (let i = 0; i < numberOfFaces; i++) {
            this.#faces[i] = Number(i + 1);
        }
        this.#lastDrawnFace = null;
    }

    shuffle() {
        assert(!!this.#faces.length, 'Error: Cannot shuffle a Dice without faces');
        return this.#lastDrawnFace = this.#faces[getRandomInt(0, this.#faces.length)];
    }

    getLastDrawnFace(){
        assert(!!this.#lastDrawnFace, 'Error: No lastDrawnFace attribute of Dice');
        return this.#lastDrawnFace;
    }
}

module.exports = Dice;